compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/ExpressionsConverter.kt

```diff
@@ -825,7 +825,7 @@ class ExpressionsConverter(
         return buildFunctionCall {
             source = arrayAccess.toFirSourceElement()
             calleeReference = buildSimpleNamedReference {
-                source = this@buildFunctionCall.source
+                source = this@buildFunctionCall.source?.fakeElement(FirFakeSourceElementKind.ArrayAccessNameReference)
```

Comment:
Because you're on this, could you check why `RawFirBuilder` set different source for `GET` and `SET` operator? In particular, why `SET` set the parent's source?

https://github.com/JetBrains/kotlin/blob/master/compiler/fir/raw-fir/psi2fir/src/org/jetbrains/kotlin/fir/builder/RawFirBuilder.kt#L1958

Reply:
Thanks! I will make the source of `FirFuntioinCall` referring to parent for set but set the source of all the `calleeReference` to the corresponding `KtArrayAccessExpression`.